logdna/resource_alert.go

```diff
@@ -92,16 +91,18 @@ func resourceAlert() *schema.Resource {
 							},
 						},
 						"immediate": {
-							Type:     schema.TypeString,
+							Type:     schema.TypeBool,
 							Optional: true,
```

Comment:
just a quick note here. Because we do not allow both terminal and immediate to be false for alerts, if terminal is provided as false, validation error will say immediate is required and immediate must be true. It's a special case.

Reply:
@ydshah2 what would be the full logic for determining what I should default terminal and immediate as?

that if terminal is provided as false => immediate has to be true?

Reply:
tried to capture the full logic here: https://github.com/logdna/terraform-provider-logdna/pull/12/files#diff-08a14d5f0eb1698501ce9e746f98a656a3eacda445356e06f8216bfb0d70e2fdR87

Reply:
Not sure where you meant. hmmm I don't know if this can be captured. I supposed they can be kept as optional fields and if the API returns an error, can be shown back to the user.